packages/mui-lab/src/Masonry/Masonry.js

```diff
@@ -96,17 +97,23 @@ export const getStyle = ({ ownerState, theme }) => {
       spacing = propValue;
     }
 
+    const baseHeight = ownerState.maxColumnHeight ?? ownerState.defaultHeight ?? undefined;
+
     return {
       margin: `calc(0px - (${spacing} / 2))`,
       '& > *': {
         margin: `calc(${spacing} / 2)`,
       },
-      ...(ownerState.maxColumnHeight && {
+      ...(baseHeight && {
         height:
           typeof spacing === 'number'
-            ? Math.ceil(ownerState.maxColumnHeight + parseToNumber(spacing))
-            : `calc(${ownerState.maxColumnHeight}px + ${spacing})`,
+            ? Math.ceil(baseHeight + parseToNumber(spacing))
+            : `calc(${baseHeight}px + ${spacing})`,
       }),
+      ...(!ownerState.maxColumnHeight &&
+        !ownerState.isSSR && {
```

Comment:
The `isSSR` prop is not truly indicating that it is not server side rendered, for e.g. disable JavaScript and open https://deploy-preview-38427--material-ui.netlify.app/material-ui/react-masonry/#basic-masonry - all demos are empty with big height (as the elements are hidden using opacity), then when the JavaScript enables, there is a layout shift in terms of height and elements appearing.